static/js/components/forms/CouponDeactivateForm.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+import React from "react";
+import { Formik, Field, Form, ErrorMessage } from "formik";
+import * as yup from "yup";
+
+import FormError from "../../components/forms/elements/FormError";
+
+type CouponDeactivateFormProps = {
+  onSubmit: Function,
+};
+
+const couponValidations = yup.object().shape({
+  coupons: yup
+    .string()
+    .required("At least one coupon name or code is required")
+    .matches(/^[\w\n]+$/, "Only letters, numbers, and underscores allowed"),
+});
+
+export const CouponDeactivateForm = ({
+  onSubmit,
+}: CouponDeactivateFormProps) => (
+  <Formik
+    onSubmit={onSubmit}
+    validationSchema={couponValidations}
+    initialValues={{
+      coupons: "",
+    }}
+    render={({ isSubmitting }) => (
+      <Form className="coupon-form">
+        <div>
+          <div className="block text-area-div">
+            <label htmlFor="coupons">
+              Coupon Codes/Names (each coupon at a separate line)*
+              <p className="small-text warning">
+                Note: Adding a coupon name will deactivate all coupon codes
+                associated with that name.
```

Comment:
I don't really understand this. Can we give an example?

Reply:
The interface allows for the deactivation of coupons using either a `coupon code` or a `coupon payment name`.

<img width="988" alt="Screenshot 2024-10-08 at 9 47 12 PM" src="https://github.com/user-attachments/assets/ee8db846-c5ed-4ac2-8625-f0449cf55597">


Deactivating by coupon code is straightforward. It will deactivate the specific coupon associated with the provided code. For example, entering `final_test2_code` in the field will deactivate that coupon.

If you enter a coupon payment name, such as `final_test_1`, it will deactivate all the coupons linked to that payment name, for instance, five coupons associated with it.

Reply:
@pdpinch added the example in the Note:

<img width="1153" alt="Screenshot 2024-10-10 at 1 13 31 PM" src="https://github.com/user-attachments/assets/6af76aa5-559e-4bc3-a93d-4f112a45235f">